frontend/src/components/navigation/NavBar.js

```diff
@@ -1,125 +1,283 @@
-import { getUserInfo } from "@/selectors/authenticationSelectors";
+import { getUserInfo, getKeycloak } from "@/selectors/authenticationSelectors";
+import { logoutUser } from "@/actions/authenticationActions";
 import * as router from "@/constants/routes";
 import * as Strings from "@/constants/strings";
+import * as Styles from "@/constants/styles";
 import * as Permission from "@/constants/permissions";
-import { LOGO, ADMIN, MINE } from "@/constants/assets";
-import Logout from "../authentication/Logout";
+import {
+  LOGO,
+  ADMIN,
+  MINE,
+  HAMBURGER,
+  LOGOUT_WHITE,
+  LOGOUT,
+  TEAM,
+  CLOSE,
+} from "@/constants/assets";
 import AuthorizationWrapper from "@/components/common/wrappers/AuthorizationWrapper";
 
 /**
- * @class NavBar - Simple fixed navbar at the top of the screen with home button/username/logout
+ * @class NavBar - fixed and responsive navigation
  */
 
 const propTypes = {
   userInfo: PropTypes.shape({ preferred_username: PropTypes.string.isRequired }).isRequired,
   activeButton: PropTypes.string.isRequired,
+  isMenuOpen: PropTypes.bool.isRequired,
+  logoutUser: PropTypes.func.isRequired,
+  toggleHamburgerMenu: PropTypes.func.isRequired,
+  keycloak: { logout: PropTypes.func.isRequired }.isRequired,
 };
 
 export class NavBar extends Component {
   menu = (
     <Menu>
       <Menu.Item key="0">
-        <Logout />
+        <button type="button" onClick={this.handleLogout}>
+          <img alt="Logout" src={LOGOUT} className="menu__img" />
+          Logout
+        </button>
       </Menu.Item>
     </Menu>
   );
 
-  render() {
-    return (
-      <div className="menu">
-        <Link
-          to={router.MINE_HOME_PAGE.dynamicRoute({
-            page: Strings.DEFAULT_PAGE,
-            per_page: Strings.DEFAULT_PER_PAGE,
-          })}
-        >
-          <img alt="Home" className="menu__img" src={LOGO} />
+  handleLogout = () => {
+    this.props.keycloak.logout();
+    localStorage.removeItem("jwt");
+    this.props.logoutUser();
+  };
+
+  renderFullNav = () => (
+    <div className="inline-flex">
+      <AuthorizationWrapper inDevelopment>
+        {/* temporary style needed only because AuthWrapper only accepts one child element */}
+        <span style={{ height: "100%" }}>
+          <Link
+            to={router.MINE_HOME_PAGE.dynamicRoute({
+              page: Strings.DEFAULT_PAGE,
+              per_page: Strings.DEFAULT_PER_PAGE,
+            })}
+          >
+            <Button
+              id={
+                includes(this.props.activeButton, router.MINE_HOME_PAGE.route)
+                  ? "active-mine-btn"
+                  : ""
+              }
+              className="menu__btn--link"
+            >
+              <img
+                alt="Mine"
+                className="padding-small--right"
+                style={{ verticalAlign: "-0.125em" }}
+                src={MINE}
+              />
+              Mines
+            </Button>
+          </Link>
```

Comment:
I don't know if this would be better, but I'll suggest it since you mentioned the duplication issue. You could create a pure function that returns this JSX. Something like:
```javasctipt
const menuButton = (activeButton, { link, button, image}) => (
<Link
            to={link.to}
          >
            <Button
              id={includes(activeButton, button.route) ? "active-mine-btn" : ""}
              className={button.className || ""}
            >
              <img
                alt={image.alt}
                className={image.className || ""}
                style={image.style}
                src={image.src}
              />
              {image.label}
            </Button>
          </Link>
)
```
and call it like:
```javascript
<AuthorizationWrapper permission={Permission.ADMIN}>
{menuButton(this.props.activeButton, {
  link: {to: router.ADMIN_DASHBOARD.route},
  button: {route: router.CONTACT_HOME_PAGE.route},
  image: {alt: "team", className: "padding-small--right icon-sm", src: TEAM, style: { verticalAlign: "-0.125em" }, label: "Admin"}
})}
// other stuff
</AuthorizationWrapper>
```

Reply:
Great input. Thank you Nathan! 

Will keep this abstraction in mind when the navigation grows larger than it currently is.